### PR TITLE
Dev/python algorithm bindings

### PR DESCRIPTION
**Pull request by as6520**
Prototype python bindings for vital abstract algorithm

1. Created python based algorithm bindings for vital::algo::image_object_detector called ImageObjectDetector
2. These bindings can be inherited by an arrow to override detect and other methods provided by algorithm
3. The arrow is registered using __vital_register_algorithm__ method and is detected by plugin explorer
4. The arrow can be used in core image_object_detector_process in pipeline runner as demonstrated in an example pipeline in examples/example_pydetector_on_image.pipe.in

### CODE REVIEW

**Comment by mattdawkins** [comment]
I guess one comment is the mixing of case styles, with some object names being UpperCamelCase some functions being lowerCamelCase and then most other functions being snake_case. Are the python object names being UpperCamelCase more desired than just making everything be snake_case like most of the rest of kwiver? Is that a python standard?

**Comment by as6520** [comment]
PEP 8 recommends class names to follow UpperCamelCase (CapWord convention) and modules be lowercase (with underscore). snake_case is used to denote an instance of an object. 

Another reason to use UpperCamelCase for python would be to prevent registration conflicts for arrows that represent the same implementation in different language.

**Comment by Purg** [comment]
Not sure if introduced in this set of changes yet, but the built/installed python module does not have a top-level __init__.py, preventing it from being imported.

File: vital/bindings/python/vital/CMakeLists.txt [hunk 1]
```diff
@@ -8,13 +8,15 @@ kwiver_add_python_module( ${CMAKE_CURRENT_SOURCE_DIR}/apm.py
   vital
   apm )
 
-kwiver_add_python_module( ${CMAKE_CURRENT_SOURCE_DIR}/__init__.py
+kwiver_add_python_module( ${CMAKE_CURRENT_SOURCE_DIR}/vital_logging.py
   vital
-  __init__ )
```
Comment (Purg) [inline vital/bindings/python/vital/CMakeLists.txt]: Seems like the ``__init__.py`` file was removed from installation here. This needs to get re-added.

File: arrows/python/simple_image_detector.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from __future__ import print_function
+
+from vital.algo import ImageObjectDetector
+from vital.types import DetectedObjectSet, DetectedObject, BoundingBox
+from sprokit.pipeline import config
+
+class SimpleImageObjectDetector(ImageObjectDetector):
```
[the remark this answers is not in the document]
Comment (as6520) [reply arrows/python/simple_image_detector.py]: My intent was to replicate [example_detector](https://github.com/Kitware/kwiver/blob/master/arrows/core/example_detector.h) in python and provide a simple use cases. I'll add documentation to make the intent clear

**Comment by Purg** [comment]
I can now run plugin_explorer and see the python implemented algorithm without sprokit enabled. LGTM after the last couple comments.

**Comment by Purg** [comment]
@linus-sherrill Do we need to do anything else with this branch before merging?

**Comment by Purg** [comment]
Jenkins test this please

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by dstoup** [comment]
Sadly this PR is causing a failure to generate from CMake on Windows. I am getting the error ..

Configuring done
CMake Error at CMake/utils/kwiver-utils-targets.cmake:175 (add_library):
  No SOURCES given to target: sprokit_python_util
Call Stack (most recent call first):
  sprokit/src/sprokit/python/util/CMakeLists.txt:13 (kwiver_add_library)

Can someone have a look?

**Comment by dstoup** [comment]
Also, there are a bunch of CentOS tests failing now that weren't yesterday.

**Pull request merged**